### PR TITLE
Fix/ping 4148 reply prompt

### PR DESCRIPTION
**Pull request by fajarism**
# Helio PR Checklist

## Please review all of the following checks before making PR

- [x] Jika menambahkan sebuah komponen, apakah sudah dipastikan tidak ada komponen yang redundant, baik secara UI maupun fungsional?
- [x] Periksa apakah komponen yang dibuat reusable (contoh: membuat komponen View untuk membuat Divider antar section)
- [x] Periksa untuk penggunaan state management yang baik dan benar (contoh: Context, useState), terutama penggunaan local state untuk optimistic UI (Pengoperan state dan data melalui props dari parent ke children dan sebaliknya)
- [x] Pastikan komponen yang dibuat sudah sesuai dengan module yang ada terutama penamaan dan penempatan untuk meningkatkan readability dan maintainability.
- [x] Implementasi fallback error untuk menangani kesalahan tak terduga, mencegah aplikasi mengalami crash
- [x] Evaluasi kompleksitas kode dan perbaiki logika yang kompleks dengan memecahnya menjadi fungsi atau komponen yang lebih kecil dan lebih mudah dimengerti.
- [x] Apakah sudah resolve semua komentar dari CodeRabbit🐇?
- [x] Apakah semua unit tests sudah Pass/Success? (Tidak ada bypass)
- [x] Periksa semua perubahan teks Bahasa Inggris approved oleh Bastian?
- [x] Apakah sudah menggunakan normalized.dimen/ normalizeFontSize untuk semua perubahan style?
- [x] Pastikan untuk mengikuti [guidelines pembuatan komponen Guidelines About Loading and Layout](https://docs.google.com/document/d/1GHbvEtFrQmEQXYBV9dLrPI654n2DbPwS1qjedajLSiw/edit)

### Summary by CodeRabbit

- New Feature: Introduced a new field `message_type` to the ChatSchema. This allows for better categorization and handling of different types of chat messages.
- Refactor: Implemented a new function `isSavingAttachmentAllowed` to determine if attachments can be saved based on the message type. This enhances security by preventing attachment saving for certain message types like `MESSAGE_TYPE_REPLY_PROMPT` and `MESSAGE_TYPE_DELETED`.

### CODE REVIEW

**Comment by bastian-purrer** [comment]
Image description   CodeRabbit



Currently reviewing new changes in this PR...

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between 4dbd308047cb5e20f789dd3791f1fc54fe696b34 and 4a845112e14a468f9e4aef9e9709672c729be942 commits.
</details>

<details>
<summary>Files selected (2)</summary>

* __tests__/database/schema/ChatSchema.test.ts (1)
* src/database/schema/ChatSchema.ts (6)
</details>






---

File: src/database/schema/ChatSchema.ts [hunk 1]
```diff
@@ -4,7 +4,11 @@ import {v4 as uuid} from 'uuid';
 import BaseDbSchema from './BaseDbSchema';
 import ChannelListSchema from './ChannelListSchema';
 import UserSchema from './UserSchema';
-import {DELETED_MESSAGE_TEXT, MESSAGE_TYPE_DELETED} from '../../utils/constants';
+import {
+  DELETED_MESSAGE_TEXT,
+  MESSAGE_TYPE_DELETED,
+  MESSAGE_TYPE_REPLY_PROMPT
+} from '../../utils/constants';
```
Comment (bastian-purrer) [inline src/database/schema/ChatSchema.ts:11]: Image description   CodeRabbit

The import statement has been modified to include a new constant `MESSAGE_TYPE_REPLY_PROMPT`. This change is necessary for the logic introduced in the function `isSavingAttachmentAllowed`.
```diff
- import {DELETED_MESSAGE_TEXT, MESSAGE_TYPE_DELETED} from '../../utils/constants';
+ import {
+   DELETED_MESSAGE_TEXT,
+   MESSAGE_TYPE_DELETED,
+   MESSAGE_TYPE_REPLY_PROMPT
+ } from '../../utils/constants';
```

File: src/database/schema/ChatSchema.ts [hunk 2]
```diff
@@ -270,7 +274,7 @@ class ChatSchema implements BaseDbSchema {
       console.log(e);
     }
     try {
-      if (json?.message?.message_type !== 'reply_prompt') {
+      if (isSavingAttachmentAllowed(json?.message?.message_type)) {
```
Comment (bastian-purrer) [inline src/database/schema/ChatSchema.ts:277]: Image description   CodeRabbit

The condition for saving attachments has been refactored into a separate function `isSavingAttachmentAllowed` which checks if the message type allows saving attachments. This improves code readability and maintainability.
```diff
- if (json?.message?.message_type !== 'reply_prompt') {
+ if (isSavingAttachmentAllowed(json?.message?.message_type)) {
```

File: src/database/schema/ChatSchema.ts [hunk 3]
```diff
@@ -319,7 +323,7 @@ class ChatSchema implements BaseDbSchema {
       console.log(e);
     }
     try {
-      if (json?.message_type !== 'reply_prompt') {
+      if (isSavingAttachmentAllowed(json?.message_type)) {
```
Comment (bastian-purrer) [inline src/database/schema/ChatSchema.ts:326]: Image description   CodeRabbit

Similar to the previous comment, the condition for saving attachments has been refactored into a separate function `isSavingAttachmentAllowed`.
```diff
- if (json?.message_type !== 'reply_prompt') {
+ if (isSavingAttachmentAllowed(json?.message_type)) {
```

File: src/database/schema/ChatSchema.ts [hunk 4]
```diff
@@ -355,7 +359,7 @@ class ChatSchema implements BaseDbSchema {
...
```
Comment (bastian-purrer) [inline src/database/schema/ChatSchema.ts:362]: Image description   CodeRabbit

Again, the condition for saving attachments has been refactored into a separate function `isSavingAttachmentAllowed`.
```diff
- if (json?.message_type !== 'reply_prompt') {
+ if (isSavingAttachmentAllowed(json?.message_type)) {
```

File: src/database/schema/ChatSchema.ts [hunk 5]
```diff
@@ -424,7 +428,9 @@ class ChatSchema implements BaseDbSchema {
       console.log(e);
     }
     try {
-      attachmentJson = JSON.stringify(data?.message?.attachments);
+      if (isSavingAttachmentAllowed(data?.message?.message_type)) {
+        attachmentJson = JSON.stringify(data?.message?.attachments);
+      }
```
Comment (bastian-purrer) [inline src/database/schema/ChatSchema.ts:433]: Image description   CodeRabbit

The condition for saving attachments has been refactored into a separate function `isSavingAttachmentAllowed`. This change also adds an additional check before stringifying the attachments, which can prevent potential errors when `data?.message?.attachments` is undefined or not valid JSON.
```diff
- attachmentJson = JSON.stringify(data?.message?.attachments);
+ if (isSavingAttachmentAllowed(data?.message?.message_type)) {
+   attachmentJson = JSON.stringify(data?.message?.attachments);
+ }
```

File: src/database/schema/ChatSchema.ts [hunk 6]
```diff
@@ -519,4 +525,8 @@ class ChatSchema implements BaseDbSchema {
   };
 }
 
+function isSavingAttachmentAllowed(messageType): boolean {
+  return ![MESSAGE_TYPE_REPLY_PROMPT, MESSAGE_TYPE_DELETED].includes(messageType);
+}
```
Comment (bastian-purrer) [inline src/database/schema/ChatSchema.ts:530]: Image description   CodeRabbit

The function `isSavingAttachmentAllowed` has been added. This function checks if the message type allows saving attachments, which improves code readability and maintainability.
```diff
+ function isSavingAttachmentAllowed(messageType): boolean {
+   return ![MESSAGE_TYPE_REPLY_PROMPT, MESSAGE_TYPE_DELETED].includes(messageType);
+ }
```

**Review by bastian-purrer — COMMENTED** [review]
Image description   CodeRabbit

<details>
<summary>Commits</summary>
Files that changed from the base of the PR and between 4dbd308047cb5e20f789dd3791f1fc54fe696b34 and 4a845112e14a468f9e4aef9e9709672c729be942 commits.
</details>

<details>
<summary>Files selected (2)</summary>

* __tests__/database/schema/ChatSchema.test.ts (1)
* src/database/schema/ChatSchema.ts (6)
</details>







<details>
<summary>Files skipped from review due to trivial changes (1)</summary>

* __tests__/database/schema/ChatSchema.test.ts

</details>

<details>
<summary>Review comments generated (6)</summary>

* Review: 6
* LGTM: 0

</details>

---

<details>
<summary>Tips</summary>

### Chat with Image description  CodeRabbit Bot (`@coderabbitai`)
- Reply on review comments left by this bot to ask follow-up questions. A review comment is a comment on a diff or a file.
- Invite the bot into a review comment chain by tagging `@coderabbitai` in a reply.

### Code suggestions
- The bot may make code suggestions, but please review them carefully before committing since the line number ranges may be misaligned. 
- You can edit the comment made by the bot and manually tweak the suggestion if it is slightly off.

### Pausing incremental reviews
- Add `@coderabbitai: ignore` anywhere in the PR description to pause further reviews from the bot.

</details>

**Review by alamsyaah — APPROVED** [review]

**Pull request merged**